### PR TITLE
F#src 5598 fix pedal driver

### PR DESCRIPTION
**Pull request by giusebar**
Updated the pyusb pedal driver. This works with the newest pedals but does not work with older pedal.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: sr_pedal/src/sr_pedal/sr_triple_pedal_pyusb.py [hunk 1]
```diff
@@ -44,10 +52,14 @@ def run(self):
...
-                    print(value)
-                except:
+                    value = self.device.read(self.endpoint_in, 512, 100)
+                    self.left_pressed = CONST_LEFT_PEDAL_VALUE in value
+                    self.middle_pressed = CONST_MIDDLE_PEDAL_VALUE in value
+                    self.right_pressed = CONST_RIGHT_PEDAL_VALUE in value
+                    if CONST_RIGHT_LEFT_PEDAL_VALUE in value:
+                        self.left_pressed = True
+                        self.right_pressed = True
+                except usb.core.USBError:
```
Comment (mikramarc) [inline sr_pedal/src/sr_pedal/sr_triple_pedal_pyusb.py]: should we at least inform user that this is happenign?

File: sr_pedal/src/sr_pedal/sr_triple_pedal_pyusb.py [hunk 2]
```diff
@@ -86,6 +99,10 @@ def publish(self, time=None):
         self.message.right_pressed = self.right_pressed
         self.publisher.publish(self.message)
 
+    def find_endpoint_in(self, device):
+        for cfg in device:
```
Comment (mikramarc) [inline sr_pedal/src/sr_pedal/sr_triple_pedal_pyusb.py]: why loops if you are returning right away?
This seems equivalent to just
return device[0][0][0].bENdpointAddress

**Comment by ethanfowler** [comment]
Does the node survive the pedal USB connection going away and coming back?

File: sr_pedal/src/sr_pedal/sr_triple_pedal_pyusb.py [hunk 3]
```diff
@@ -56,7 +68,8 @@ def connect(self):
         rospy.loginfo("Waiting for pedal...")
         while ((not rospy.is_shutdown()) and self.device is None):
             self.device = usb.core.find(idVendor=self.vendor_id, idProduct=self.product_id)
-            self.publish()
+            self.device.reset()
+            self.endpoint_in = self.find_endpoint_in(self.device)
```
Comment (mikramarc) [inline sr_pedal/src/sr_pedal/sr_triple_pedal_pyusb.py]: maybe find_endpoint_in_device?

**Review by mikramarc — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Review by ethanfowler — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**